### PR TITLE
Issue 123: Specify the project main module

### PR DESCRIPTION
**Pull request by OldSneerJaw**
Set the sync-function-loader module as the primary entry point (aka the main module) for the npm project so that a dependent project can generate a sync function programmatically by invoking `require('synctos').load('/path/to/my-doc-definitions.js')`.

Addresses issue #123.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -249,7 +249,7 @@ An example of a mix of channel and role access assignments:
...
+* `allowAttachments`: (optional) Whether to allow the addition of [file attachments](http://developer.couchbase.com/documentation/mobile/current/references/sync-gateway/rest-api/index.html#!/attachment/put_db_doc_attachment) for the document type. In addition to a static value (e.g. `allowAttachments: true`), this property may also be assigned a value dynamically via a function (e.g. `allowAttachments: function(doc, oldDoc) { ... }`) where the parameters are as follows: (1) the document (if deleted, the `_deleted` property will be `true`) and (2) the document that is being replaced (if any; it will be `null` if it has been deleted or does not exist). Defaults to `false` to prevent malicious/misbehaving clients from polluting the bucket/database with unwanted files. See the `attachmentConstraints` property and the `attachmentReference` validation type for more options.
```
Comment (OldSneerJaw) [inline README.md]: The only change here was to fix the URL to the file attachments documentation on Couchbase's site.

File: README.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ Synctos: The Syncmaker. A utility to aid with the process of designing well-stru
...
 
-To learn more about Sync Gateway, check out [Couchbase](http://www.couchbase.com/)'s comprehensive [developer documentation](http://developer.couchbase.com/documentation/mobile/current/get-started/sync-gateway-overview/index.html). And, for a comprehensive introduction to synctos, see the post [Validating your Sync Gateway documents with synctos](https://blog.couchbase.com/validating-your-sync-gateway-documents-with-synctos/) on the official Couchbase blog.
+To learn more about Sync Gateway, check out [Couchbase](http://www.couchbase.com/)'s comprehensive [developer documentation](http://developer.couchbase.com/documentation/mobile/current/guides/sync-gateway/index.html). And, for a comprehensive introduction to synctos, see the post [Validating your Sync Gateway documents with synctos](https://blog.couchbase.com/validating-your-sync-gateway-documents-with-synctos/) on the official Couchbase blog.
```
Comment (OldSneerJaw) [inline README.md]: Fixed this broken link to Couchbase's documentation.

File: make-sync-function [hunk 3]
```diff
@@ -20,7 +20,7 @@ var outputFilename = process.argv[3];
 
 var syncFunc;
 try {
-  syncFunc = syncFunctionLoader.load(syncDocDefnFilename, scriptDir);
+  syncFunc = syncFunctionLoader.load(syncDocDefnFilename);
```
Comment (ziemek) [inline make-sync-function]: Minor: now that `scriptDir` is no longer referenced, its definition can be removed from this file.

Comment (OldSneerJaw) [inline make-sync-function]: Right you are. I will remove the variable's declaration.

File: etc/test-helper.js [hunk 4]
```diff
@@ -1,11 +1,3 @@
-var validationErrorFormatter = require('./validation-error-message-formatter.js');
```
Comment (OldSneerJaw) [inline etc/test-helper.js]: This change is not directly related to the primary purpose of this pull request, but it only involves moving this code further down the file so it seemed unnecessary to create a new issue and PR just for it.

File: package.json [hunk 5]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "synctos",
   "version": "1.9.1",
+  "main": "etc/sync-function-loader.js",
   "dependencies": { },
   "devDependencies": {
     "expect.js": "^0.3.1",
     "jshint": "^2.9.4",
-    "mocha": "^3.3.0",
+    "mocha": "^3.4.1",
```
Comment (OldSneerJaw) [inline package.json]: The updated version of mocha is not directly related to this pull request's purpose.

**Comment by OldSneerJaw** [comment]
Ready for another review.

**Comment by ziemek** [comment]
Looks good! 👍
